src/pages/dashboardGroupEdit/components/GroupForm.tsx

```diff
@@ -293,12 +293,12 @@ export default function GroupForm({
   });
 
   useEffect(() => {
-    if (formik.dirty) {
+    if (formik.dirty || productSelected) {
```

Comment:
this way, a group opened in editMode will always be blocked when exit even if there are not changes!